packages/plugin-aws-lambda/src/index.js

```diff
@@ -0,0 +1,84 @@
+const bugsnagInFlight = require('@bugsnag/in-flight')
+
+const BugsnagPluginAwsLambda = {
+  name: 'awsLambda',
+
+  load (client) {
+    bugsnagInFlight.trackInFlight(client)
+
+    return {
+      createHandler ({ flushTimeoutMs = 2000 } = {}) {
+        return wrapHandler.bind(null, client, flushTimeoutMs)
+      }
+    }
+  }
+}
+
+function wrapHandler (client, flushTimeoutMs, handler) {
+  let _handler = handler
+
+  if (handler.length > 2) {
+    // This is a handler expecting a 'callback' argument, so we convert
+    // it to return a Promise so '_handler' always has the same API
+    _handler = promisifyHandler(handler)
+  }
+
+  return async function (event, context) {
+    client.addMetadata('AWS Lambda context', context)
+
+    try {
+      return await _handler(event, context)
+    } catch (err) {
+      if (client._config.autoDetectErrors && client._config.enabledErrorTypes.unhandledExceptions) {
```

Comment:
Since we're async-ly `try`ing user code, doesn't this mean we'll catch promise rejections too, preventing them from bubbling up to the `onunhandledrejection` handler?

In that case we'll fail to honour the `enabledErrorTypes.unhandledRejections` configuration. And it also means the `severityReason.type` will always be `unhandledException`.

I'm not quite sure what we should do to "fix" this, because really with the `async` keyword, the concept of unhandled exceptions/rejections are somewhat muddied.